cryptoutil/cryptoutil.go

```diff
@@ -0,0 +1,54 @@
+// Package cryptoutil provides a set of functions for 2-way encryption and hashing.
+package cryptoutil // import "github.com/teamwork/utils/cryptoutil"
```

Comment:
Maybe just `crypt`?

Reply:
Bit morbid (haha) but sure, I thought I'll use `crypto` since thats the name of the Go package though 😄

Reply:
The general naming scheme is (from the README):

 > The naming scheme is `[type]util` or `[pkgname]util`. If there already is a `*util` packge in stdlib it's named `utilx` (e.g. `ioutilx`).

Since his is enhancing the `crytpto/aes` package it should be named `aesutil` IMHO. This will allow us to add other crypto-related stuff without turning this in to a "random bag of functions" kind of package.

Reply:
Aye, but in the future if we add a util function for hashing, which is very likely, would we make a... sha1util? and an md5util? 😄 I still think `cryptoutil` is best since it sticks to our convention and doesn't restrict it to a single encryption/hashing algorithm, at least from its name. 😄

Reply:
It seems to me that a hashing function is completely unrelated to encrypting something with e.g. AES.

If we want to have a new `sha256util` (both SHA1 and MD5 are pretty broken cryptographically btw) then I don't see a problem with many small packages. From the caller's perspective it seems to look much nicer to me:

    cryptoutil.HashString("foo")

vs:

    sha256util.HashString("foo")

The second has a lot more info about what it does (hash a string as sha256), while the former doesn't. Sure, there are function comments, but if we can make something clear without a function comment then that's better.

Just my 2 cents anyway.

Reply:
Speaking of morbid btw, this is why I always name my exceptions `dead` in Python or Ruby, so you can use `raise dead` :-)

Reply:
Yeah I can see where you're coming from, I guess `aesutil.Encrypt` makes it more obvious at a glance 😄

Reply:
Also – I just remembered this – in Desk we added an `Encrypt()` and `Decrypt()` based on nacl: https://github.com/Teamwork/desk/blob/master/modules/crypto/crypto.go#L22

This is, I think, a better approach than using AES directly. For example the Go docs say:


	// It's important to remember that ciphertexts must be authenticated
	// (i.e. by using crypto/hmac) as well as being encrypted in order to
	// be secure.

And I think there may be some other caveats as well.

nacl is designed to think about these things "for you", and should be more secure.

Reply:
Aye, my main concern though would be backwards compatibility, for example we have tokens encrypted with AES in Projects codebase which are later decrypted in launchpad, so I'm not sure in this case it would work. 😕